x-pack/legacy/plugins/infra/public/components/logging/log_text_stream/loading_item_view.tsx

```diff
@@ -6,20 +6,36 @@
 
 /* eslint-disable max-classes-per-file */
 
-import { EuiButtonEmpty, EuiIcon, EuiProgress, EuiText } from '@elastic/eui';
-import { FormattedMessage, FormattedRelative } from '@kbn/i18n/react';
+import {
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiLoadingSpinner,
+  EuiButton,
+} from '@elastic/eui';
+import { FormattedMessage, FormattedTime } from '@kbn/i18n/react';
 import * as React from 'react';
+import { Unit } from '@elastic/datemath';
 
 import euiStyled from '../../../../../../common/eui_styled_components';
+import { LogTextSeparator } from './log_text_separator';
+import { extendDatemath } from '../../../utils/datemath';
+
+type Position = 'start' | 'end';
```

Comment:
Would it be overkill to make this an `enum`? Saves typing to make it string literals but maybe it's easier autocomplete as enum constants.

Reply:
I'd prefer to keep it as a union. The language server is smart enough to only suggest those two options, and it prevents consumers from having to `import` the enum to use the component.

It also keeps the lines shorter in general :)

<img width="540" alt="Screenshot 2020-02-20 at 14 06 13" src="https://user-images.githubusercontent.com/57448/74936374-4e570580-53ea-11ea-8aaf-170282b62e60.png">

Reply:
yay for unions :+1: enums are harder to integrate with io-ts and offer few benefits if no specific key-value mapping is required. On the contrary, they tend to promote unsafe type assertions when crossing API boundaries as compared to plain string literals.